src/Pact/Eval.hs

```diff
@@ -233,6 +233,28 @@ evalNamespace info setter m = do
         Nothing -> ModuleName nn (Just n)
         Just {} -> mn
 
+-- | Lookup module in state or database with exact match on 'ModuleName'.
+lookupModule :: HasInfo i => i -> ModuleName -> Eval e (Maybe (ModuleData Ref))
+lookupModule i mn = do
+  loaded <- preuse $ evalRefs . rsLoadedModules . ix mn
+  case loaded of
+    Just (m,_) -> return $ Just m
+    Nothing -> do
+      stored <- readRow (getInfo i) Modules mn
+      case stored of
+        Just mdStored -> do
+          natives <- view $ eeRefStore . rsNatives
+          let natLookup (NativeDefName n) = case HM.lookup (Name n def) natives of
+                Just (Direct t) -> Just t
+                _ -> Nothing
```

Comment:
should this error here if we can't find a native?

Reply:
`fromPersistDirect` creates an error on `Nothing`, thus the `Either` result below